Manager/SubscriberManager.php

```diff
@@ -0,0 +1,194 @@
+        $result = $this->subscriberWebService->InsertSubscriber(new InsertSubscriber($mbzSubscriber, true));
+        $subscriber->setId($result->getInsertSubscriberResult());
+
+        $this->logger->info('New subscriber inserted in Mindbaz', ['id' => $subscriber->getId()]);
+    }
+
+    /**
+     * @param Subscriber $subscriber
+     */
+    public function unsubscribe(Subscriber $subscriber)
+    {
+        $result = $this->subscriberWebService->Unsubscribe(new Unsubscribe($subscriber->getId(), null, null));
+        if (true === $result->getUnsubscribeResult()) {
+            $this->logger->info('Subscriber successfully unsubscribed', ['id' => $subscriber->getId()]);
+        } else {
+            $this->logger->error('An error occurred while unsubscribing subscriber', ['id' => $subscriber->getId(), 'response' => $result->getUnsubscribeResult()]);
+        }
+    }
+
+    /**
+     * @param array $emails
+     *
+     * @return array
+     */
+    public function findByEmail(array $emails)
+    {
+        $result = $this->subscriberWebService->GetSubscribersByEmail(
+            new GetSubscribersByEmail(
+                (new ArrayOfString())->setString($emails),
+                (new ArrayOfInt())->setInt([0, 1])
+            )
+        )->getGetSubscribersByEmailResult();
+
+        // Unable to find related subscribers
+        if (null === $result) {
+            return [];
+        }
+
+        $subscribers = [];
+        foreach ($result->getSubscriber() as $subscriber) {
+            $subscribers[] = $this->serializer->deserialize($subscriber, Subscriber::class, SubscriberEncoder::FORMAT);
+        }
+
+        return $subscribers;
+    }
+
+    /**
+     * @param string $email
+     *
+     * @return Subscriber|null
+     */
+    public function findOneByEmail($email)
+    {
+        $subscribers = $this->findByEmail([$email]);
+
+        return 0 < count($subscribers) ? $subscribers[0] : null;
+    }
+
+    /**
+     * @param int                 $idCampaign
+     * @param Subscriber          $subscriber
+     * @param \Swift_Mime_Message $message
+     */
+    public function send($idCampaign, Subscriber $subscriber, \Swift_Mime_Message $message)
+    {
+        $response = $this->oneshotWebService->Send(
+            new Send(
+                $idCampaign,
+                $subscriber->getId(),
+                $this->getBody($message, 'text/html'),
+                $this->getBody($message, 'text/plain'),
+                $message->getSender(),
+                $message->getSubject()
+            )
+        );
+        if (self::MINDBAZ_SEND_RESPONSE_OK === $response->getSendResult()) {
+            $this->logger->info('Message successfully sent to subscriber', ['id' => $subscriber->getId()]);
+        } else {
+            $this->logger->error('An error occurred while sending the message to subscriber', ['id' => $subscriber->getId(), 'response' => $response->getSendResult()]);
```

Comment:
yes replace logger by an exception
In a app, i could use this mailer like this (for example) :
try { $this->mailer->send($message); } catch (SendException $e) { // Update user to clean emails in database }